tests/test_deserialization.py

```diff
@@ -1315,6 +1315,77 @@ class MySchema(Schema):
         errors = MySchema(partial=True).validate({'foo': 3}, partial=('bar', 'baz'))
         assert errors == {}
 
+    def test_unknown_fields_deserialization(self):
+        class MySchema(Schema):
+            foo = fields.Integer()
+
+        data = MySchema().load({'foo': 3, 'bar': 5})
+        assert data['foo'] == 3
+        assert 'bar' not in data
+
+        data = MySchema(unknown=ALLOW).load({'foo': 3, 'bar': 5}, unknown=IGNORE)
+        assert data['foo'] == 3
+        assert 'bar' not in data
+
+        data = MySchema().load({'foo': 3, 'bar': 5}, unknown=ALLOW)
+        assert data['foo'] == 3
+        assert data['bar']
+
+        data = MySchema(unknown=ALLOW).load({'foo': 3, 'bar': 5})
+        assert data['foo'] == 3
+        assert data['bar']
+
+        with pytest.raises(ValidationError) as excinfo:
+            MySchema(unknown=ALLOW).load({'foo': 'asd', 'bar': 5})
+        assert 'foo' in str(excinfo)
+
+        data = MySchema(unknown=ALLOW, many=True).load([
+            {'foo': 1},
+            {'foo': 3, 'bar': 5}
+        ])
+        assert 'foo' in data[1]
+        assert 'bar' in data[1]
+
+        with pytest.raises(ValidationError) as excinfo:
+            MySchema(unknown=RAISE).load({'foo': 3, 'bar': 5})
+        err = excinfo.value
+        assert 'bar' in err.messages
+        assert err.messages['bar'] == ['Unknown field.']
+
+        with pytest.raises(ValidationError) as excinfo:
+            MySchema(unknown=RAISE, many=True).load([
+                {'foo': 'abc'},
+                {'foo': 3, 'bar': 5},
+            ])
+        err = excinfo.value
+        assert 0 in err.messages
+        assert 'foo' in err.messages[0]
+        assert err.messages[0]['foo'] == ['Not a valid integer.']
+        assert 1 in err.messages
+        assert 'bar' in err.messages[1]
+        assert err.messages[1]['bar'] == ['Unknown field.']
+
```

Comment:
Can you also add a test that `unknown` behaves correctly when `data_key` is set for a field?

Reply:
Well seen, it indeed doesn't work properly. I'm digging on it.

Reply:
Fixed. :+1: